Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -124,6 +120,116 @@ public override void Serialize(IIonWriter writer, object item)
             writer.StepOut();
         }
 
+        // Deserialize the given property and return bool to indicate whether the deserialized result should be used.
+        private bool TryDeserializeProperty(
+            PropertyInfo property, IIonReader reader, IonType ionType, ref object deserialized)
+        {
+            // We deserialize whether or not we ultimately use the result because values
+            // for some Ion types need to be consumed in order to advance the reader.
+            deserialized = ionSerializer.Deserialize(reader, property.PropertyType, ionType);
+            
+            if (IsReadOnlyProperty(property))
+            {
+                // property.SetValue() does not work with a readonly property.
+                // logic for handling deserializing readonly properties happens during field processing
+                // when we detect backing fields for the property.
+                return false;
```

Comment:
Reading this - will this then work? If you read the value but then continue if it's read only, then you cannot read it again when you're trying do deserialize the field.

Reply:
In .NET, properties have backing fields under the hood. So when we encounter a readonly property, since we can't call `property.SetValue()`, we instead process that property's backing field (which does not have the same readonly constraints. ie. `field.SetValue()` works).

The backing field has a distinct name from the property (ie. a property named `Major` has a backing field called `<Major>k__BackingField`) and the field will have been serialized with that distinct name so we still get a chance to find the backing field during another iteration of the loop.